### PR TITLE
feat(api): add update locker

### PR DESCRIPTION
**Pull request by kduret**
## Description

lock update to avoid multiple updates at the same time

**Fixes** MON-12296

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x
- [x] 22.10.x (master)

### CODE REVIEW

File: src/Core/Platform/Infrastructure/Repository/UpdateLockerException.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+use Centreon\Domain\Repository\RepositoryException;
+
+class UpdateLockerException extends RepositoryException
+{
+    /**
+     * @return self
+     */
+    public static function errorWhileLockingUpdate(\Throwable $e): self
+    {
+        return new self(_('Error while locking update process'), 0, $e);
```
Comment (a-launois) [inline src/Core/Platform/Infrastructure/Repository/UpdateLockerException.php]: Translations are missing for the these error messages

File: composer.json [hunk 2]
```diff
@@ -75,7 +75,8 @@
         "symfony/translation": "5.4.*",
         "symfony/validator": "5.4.*",
         "symfony/yaml": "5.4.*",
-        "psr/container": "1.1.1"
+        "psr/container": "1.1.1",
+        "symfony/lock": "5.4.*"
```
Comment (kduret) [inline composer.json]: move to alphabetical order

File: src/Core/Platform/Application/Repository/UpdateLockerRepositoryInterface.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+declare(strict_types=1);
+
+namespace Core\Platform\Application\Repository;
+
+interface UpdateLockerRepositoryInterface
+{
+    /**
+     * Lock update process
+     *
+     * @return bool if the lock has been properly acquired
```
Comment (kduret) [inline src/Core/Platform/Application/Repository/UpdateLockerRepositoryInterface.php:30]:
```suggestion
     * @return bool if the lock has been properly acquired
     *
     * @throws \Exception
```

File: src/Core/Platform/Infrastructure/Repository/UpdateLockerException.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Core\Platform\Infrastructure\Repository;
+
+use Centreon\Domain\Repository\RepositoryException;
+
+class UpdateLockerException extends RepositoryException
```
Comment (kduret) [inline src/Core/Platform/Infrastructure/Repository/UpdateLockerException.php]: move exception to Application layer

File: src/Core/Platform/Infrastructure/Repository/FsUpdateLockerRepository.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Platform\Application\Repository\UpdateLockerRepositoryInterface;
+use Symfony\Component\Lock\LockFactory;
+use Symfony\Component\Lock\LockInterface;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockExpiredException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+
+class FsUpdateLockerRepository implements UpdateLockerRepositoryInterface
```
Comment (kduret) [inline src/Core/Platform/Infrastructure/Repository/FsUpdateLockerRepository.php]:
```suggestion
class SymfonyUpdateLockerRepository implements UpdateLockerRepositoryInterface
```

File: src/Core/Platform/Application/UseCase/UpdateVersions/UpdateVersions.php [hunk 6]
```diff
@@ -73,6 +80,28 @@ public function __invoke(
...
 
+    /**
+     * Lock update process
+     */
+    private function lockUpdate(): void
+    {
+        $this->info('Locking centreon update process...');
+
+        if (!$this->updateLocker->lock()) {
+            throw new \Exception('Update already in progress');
```
Comment (kduret) [inline src/Core/Platform/Application/UseCase/UpdateVersions/UpdateVersions.php]: create exception class `UpdateVersionsException`

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16916,3 +16916,24 @@ msgstr "Une erreur s'est produite lors de la récupération des catégories d'h
...
+msgstr "Une erreur s'est produite lors de la récupération des mises à jour disponibles"
+
+msgid "An error occurred when applying update %s (%s)"
+msgstr "Une erreur s'est produite lors de l'application de la mise à jour %s (%s)"
+
+msgid "Error while locking update process"
+msgstr "Erreur lors du verouillage du processus de mise à jour"
+
+msgid "'Error while unlocking update process"
+msgstr "Erreur lors du dévérouillage du processus de mise à jour"
```
Comment (a-launois) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Erreur lors du verrouillage du processus de mise à jour"

msgid "'Error while unlocking update process"
msgstr "Erreur lors du déverrouillage du processus de mise à jour"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -16916,3 +16916,24 @@ msgstr "Une erreur s'est produite lors de la récupération des catégories d'h
 
 msgid "Warning, maximum size exceeded for input '%s' (max: %d), it will be truncated upon saving"
 msgstr "Attention, taille maximale dépassée pour le champ '%s' (max: %d), il sera tronqué à l'enregistrement"
+
+msgid "Update already in progress"
+msgstr "Mise à jour déjà en cours"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Une mise à jour est déjà en cours"
```

**Review by a-launois — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**